### ISSUE TITLE
Gateway does not close after usage, blocks script exit

### ISSUE THREAD

**Issue by bavobbr** [int128/groovy-ssh#71]
I use gateway to do ssh tunelling. This works great, but I found that my scripts using this will not exit at the end because the gateway keeps running. I tried looking for a way to signal it to close but can not find a hook into it.
My only option is to call System.exit()

Demo

```
    def service = new Service()
    service.remotes {
        gw {
            host = 'gatewaydevice'
            user = 'myuser'
            password = 'mypassword'
            knownHosts = service.settings.allowAnyHosts
        }
        target {
            host = 'targetdevice'
            user = 'myuser'
            password = 'mypassword'
            gateway = service.remotes.gw
        }
    }
    service.run {
        session(service.remotes.target) {
            execute('virsh list') {result ->
                println result
            }
        }
    }
```

This will not quit when run as Groovy script unless you kill it hard

**Comment by timend** [comment]
I'm seeing the same problem. For me it results in eventually running out of memory on the gateway host, because the application polls the status of a host via SSH  (currently using a new connection each time).

**Comment by int128** [comment]
Fixed in 2.4.3.

**Comment by bavobbr** [comment]
Tested and works perfect now, thank you!
